core/src/main/java/denominator/mock/MockAllProfileResourceRecordSetApi.java

```diff
@@ -22,17 +17,16 @@
 import static denominator.model.ResourceRecordSets.nameTypeAndQualifierEqualTo;
 import static denominator.model.ResourceRecordSets.notNull;
 
-public class MockAllProfileResourceRecordSetApi
-    implements denominator.AllProfileResourceRecordSetApi {
+class MockAllProfileResourceRecordSetApi implements denominator.AllProfileResourceRecordSetApi {
 
-  protected final Provider provider;
-  protected final SortedSet<ResourceRecordSet<?>> records;
-  protected final Filter<ResourceRecordSet<?>> filter;
+  private final Map<String, Collection<ResourceRecordSet<?>>> data;
+  private final String zoneName;
```

Comment:
had to defer expansion to the zone in order to support zone mutation in the future. otherwise, a cache of a record set api would work fine even after the zone was deleted.